CMake/SetFolder.cmake

```diff
@@ -0,0 +1,33 @@
+function(set_folder directory folder)
+  set(options)
+  set(oneValueArgs)
+  set(multiValueArgs EXCLUDE_TARGET EXCLUDE_DIRECTORY)
+  cmake_parse_arguments(SF "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+  set(exclude_directory)
+  foreach(excl IN LISTS SF_EXCLUDE_DIRECTORY)
+    get_filename_component(excl_abs "${excl}" ABSOLUTE)
+    list(APPEND exclude_directory ${excl_abs})
+  endforeach()
+
+  get_directory_property(targets DIRECTORY "${directory}" BUILDSYSTEM_TARGETS)
```

Comment:
So CMake 3.7 is required to get proper source groups in Visual Studio? Not sure how I feel about that. If someone's using pre-3.7, it's going to be non-obvious to them what they need to do to get source groups to work properly.